tests/test_failure_handler.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```

Comment:
I would normally say to remove this class since you're technically testing a private method which should never be called directly. But I think in this case it's fine. Eventually we should separate direct interaction with the api and failure handling into a separate class that this one can use

Reply:
On second thought, I do think we should remove this test class for now. We're currently double testing this logic between the two classes and this class isn't as close to what we will eventually be doing as I thought